tests/phpunit/tests/formatting/wpTrimExcerpt.php

```diff
@@ -92,4 +92,60 @@ public function test_should_generate_excerpt_for_empty_values() {
 		$this->assertSame( 'Post content', wp_trim_excerpt( null, $post ) );
 		$this->assertSame( 'Post content', wp_trim_excerpt( false, $post ) );
 	}
+
+	/**
+	 * Tests that `wp_trim_excerpt()` unhooks `wp_filter_content_tags()` from 'the_content' filter.
+	 *
+	 * @ticket 56588
+	 */
+	public function test_wp_trim_excerpt_unhooks_wp_filter_content_tags() {
+		$post = self::factory()->post->create();
+
+		/*
+		 * Record that during 'the_content' filter run by wp_trim_excerpt() the
+		 * wp_filter_content_tags() callback is not used.
+		 */
+		$has_filter = true;
+		add_filter(
+			'the_content',
+			static function( $content ) use ( &$has_filter ) {
+				$has_filter = has_filter( 'the_content', 'wp_filter_content_tags' );
+				return $content;
+			}
+		);
+
+		wp_trim_excerpt( '', $post );
+
+		$this->assertFalse( $has_filter, 'wp_filter_content_tags() was not unhooked in wp_trim_excerpt()' );
```

Comment:
Assert it gets restored:

```suggestion
		$this->assertSame( 10, has_filter( 'the_content', 'wp_filter_content_tags' ) );
```

Reply:
Granted, this is also done in `test_wp_trim_excerpt_should_not_permanently_unhook_wp_filter_content_tags`

Reply:
Exactly, related is my reply in https://github.com/WordPress/wordpress-develop/pull/4439#discussion_r1202696771.